atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/sweep/GetCellTimestamps.java

```diff
@@ -94,17 +100,21 @@ private void fetchBatchOfTimestamps() {
     private void fetchBatchOfTimestampsBeginningAtStartRow() {
         byte[] rangeStart = startRowInclusive;
 
+        ExecutorService executor = PTExecutors.newFixedThreadPool(config.sweepReadThreads());
```

Comment:
Huh, this was actually the opposite of what I thought would happen - I thought the executor service would be created and destroyed within `CqlExecutor.getTimestamps`.

I think that this will almost always be equivalent, though: `timestamps.isEmpty` will be false after one iteration except in cases where we're sweeping a thorough table and have `batchHint` rows of tombstones in front of us.

Reply:
It seemed weird that we'd be potentially recreating the executor in each `getTimestamps` call at first glance, and we had a `while` loop above. But yeah, we just do that on the multiple tombstones scenario, which is pretty rare though. I think we're fine leaving like this is well — we already start to extract some of the logic from the CqlExecutor.